### PR TITLE
Lab 3 - Letter Frequency

### LINKED ISSUE #322: Lab3 - Letter Frequency

### PR DESCRIPTION
**Pull request by jessodri**
Fixes #322 

Review of colleague's PR #

#### Changes proposed in this PR:

- Add function to count letter frequency
- Add tests

### CODE REVIEW

File: 03_letters/jessodri/main_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+func TestSortLetters(t *testing.T) {
+	testCases := []struct {
+		description string
+		letterFreq  map[rune]int
+		expected    []string
+	}{
+		{"happy path", map[rune]int{'a': 1, 'b': 2, 'c': 3, 'd': 4, 'e': 5}, []string{"a:1", "b:2", "c:3", "d:4", "e:5"}},
+		{"characters and empty spaces", map[rune]int{' ': 4, '!': 5, '&': 1, 'b': 2, 'c': 3}, []string{" :4", "!:5", "&:1", "b:2", "c:3"}},
```
Comment (golangcibot) [inline 03_letters/jessodri/main_test.go]: line is 133 characters (from `lll`)

File: 03_letters/jessodri/main_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			"aa ! && cc",
+			map[rune]int{' ': 3, 'a': 2, 'c': 2, '!': 1, '&': 2},
+		},
+		{
+			"empty",
+			"",
+			map[rune]int{},
+		},
+	}
+
```
Comment (willshen8) [inline 03_letters/jessodri/main_test.go:79]: Your code is clean and easy to understand. Well done! The only thing I would add is the completeness of test-cases: As rune represent unicode codepoints, it would be good if your test cases can go beyond letters, numbers and characters. But this is the bonus part, so I'll leave it up to you whether you want to add more or not. Cheers.

File: 03_letters/jessodri/main.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
```
Comment (rokane) [inline 03_letters/jessodri/main.go:11]: Awesome work!

File: 03_letters/jessodri/main.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	for k, v := range m {
+		s := string(k) + ":" + strconv.Itoa(v)
+		letterFreq = append(letterFreq, s)
+	}
+	sort.Strings(letterFreq)
+	return letterFreq
+}
+
+func main() {
+	fmt.Fprintln(out, sortLetters(letters("This is a string for mapping!")))
```
Comment (rokane) [inline 03_letters/jessodri/main.go:32]: Note the lab does ask to print the below out, but if your tests are covering this case I will let it go.
> fmt.Println(strings.Join(sortLetters(letters("aba")), "\n"))

**Review by rokane — APPROVED** [review]
Fantastic work!! Your code and tests are super clean, succinct and easy to follow. I agree with @willshen8 that you could add some more test cases to include the variations on runes, but since it is an optional extra I am happy with this. Good work 👍
